File: 03-encapsulamiento/biblioteca/libro.py
"""Class Libro"""

from genero import Genero


class Libro:
    def __init__(self, titulo: str, autor: str, genero: Genero, paginas: int):
        self.__titulo = titulo
        self.__autor = autor
        self.__genero = genero
        self.__paginas = paginas

    @property
    def titulo(self):
        return self.__titulo

    @property
    def paginas(self):
        return self.__paginas

    @property
    def autor(self):
        return self.__autor

    @property
    def genero(self):
        return self.__genero

    def __eq__(self, other):
        return (
            self.__titulo == other.titulo
            and self.__autor == other.autor
            and self.__genero == other.genero
            and self.__paginas == other.paginas
        )

File: workspace-2025-1c/ModeloParcial/src/parciales/Libro.java
package parciales;

public class Libro {
    private String titulo;
    private String autor;
    private String genero;
    private int cantidadDisponible;

    public Libro(String titulo, String autor, String genero, int cantidadDisponible) {
        this.titulo = titulo;
        this.autor = autor;
        this.genero = genero;
        this.cantidadDisponible = cantidadDisponible;
    }

    public String getTitulo() {
        return titulo;
    }

    public String getAutor() {
        return autor;
    }

    public String getGenero() {
        return genero;
    }

    public int getCantidadDisponible() {
        return cantidadDisponible;
    }

    @Override
    public String toString() {
        return titulo + ", " + autor + ", " + genero + ", " + cantidadDisponible;
    }
}

